Code coverage is failing

See https://github.com/quickwit-oss/quickwit/runs/5670210584?check_suite_focus=true

Here is the cryptic error:

```rust
error: no rules expected the token `aarch64_apple`
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/src/cpu.rs:257:13
    |
165 |     macro_rules! features {
    |     --------------------- when calling this macro
...
257 |             aarch64_apple: true,
    |             ^^^^^^^^^^^^^ no rules expected this token in macro call

error[E0425]: cannot find value `AES` in module `cpu::arm`
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/src/aead/aes.rs:381:65
    |
381 |         if cpu::intel::AES.available(cpu_features) || cpu::arm::AES.available(cpu_features) {
    |                                                                 ^^^ not found in `cpu::arm`
    |
help: consider importing this constant
    |
15  | use crate::cpu::intel::AES;
    |

error[E0425]: cannot find value `PMULL` in module `cpu::arm`
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/src/aead/gcm.rs:315:26
    |
315 |             || cpu::arm::PMULL.available(cpu_features)
    |                          ^^^^^ not found in `cpu::arm`

error[E0425]: cannot find value `ARMCAP_STATIC` in this scope
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/src/cpu.rs:235:41
    |
235 |             if self.mask == self.mask & ARMCAP_STATIC {
    |                                         ^^^^^^^^^^^^^ not found in this scope

```

--- a/.github/workflows/coverage.yml
+++ b/.github/workflows/coverage.yml
@@ -113,29 +113,26 @@ jobs:
 
       - uses: actions-rs/toolchain@v1
         with:
-          toolchain: stable
-          profile: minimal
+          toolchain: nightly
+          components: clippy, llvm-tools-preview, rustfmt
+
+      - name: Install cargo-llvm-cov
+        run: |
+          curl -LsSf https://github.com/taiki-e/cargo-llvm-cov/releases/latest/download/cargo-llvm-cov-x86_64-unknown-linux-gnu.tar.gz | \
+          tar xzf - -C ~/.cargo/bin
 
       - uses: Swatinem/rust-cache@v1
 
-      - name: Compile tests
-        run: cargo test --no-run --locked
+      - name: Generate code coverage
+        run: |
+          cargo +nightly llvm-cov clean --workspace
+          cargo +nightly llvm-cov --test failpoints --no-report --features fail/failpoints
+          cargo +nightly llvm-cov --no-report --all-features
+          cargo +nightly llvm-cov --no-run --lcov --output-path lcov.info
 
-      - name: Run tests
-        run: cargo test --all-features
-
-      # Code coverage. Was removed due to #1221
-
-      # - name: Generate code coverage
-      #   run: |
-      #     cargo +nightly llvm-cov clean --workspace
-      #     cargo +nightly llvm-cov --test failpoints --no-report --features fail/failpoints
-      #     cargo +nightly llvm-cov --no-report --all-features
-      #     cargo +nightly llvm-cov --no-run --lcov --output-path lcov.info
-
-      # - name: Upload coverage to Codecov
-      #   uses: codecov/codecov-action@v2.1.0
-      #   with:
-      #     token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
-      #     files: lcov.info
-      #     fail_ci_if_error: true
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2.1.0
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+          files: lcov.info
+          fail_ci_if_error: true

